docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+```
+
+Depending on the data type, DeepReg will expect the images and labels to be stored in
+specific structures: check the [data loader configuration](dataset_loader.html) for more
+details.
+
+### Labeled key - Required
+
+The `labeled` key indicates whether labels should be used during training. A Boolean is
+used to indicate the usage of labels:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  labeled: true
+```
+
+### Type key - Required
+
+The type of data loader used will depend on how one wants to train the network.
+Currently, DeepReg data loaders support the "paired", "unpaired" and "grouped" training
+strategies. Passing a string that doesn't match any of the above would raise an error.
+The data loader type would be specified using the `type` key:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  type: "paired" # one of "paired", "unpaired" or "grouped"
+```
+
+#### Data loader dependent keys
+
+Depending on which string is passed to the `type` key, DeepReg will initialise a
+different data loader instance with different sampling strategies. These are reviewed in
+depth in the [dataset loader configuration](dataset_loader.html) documentation. Here we
+outline the arguments necessary to configure the different dataloaders.
+
+###### Sample_label - Required
+
+In the case that we have more than one label per image, we need to indicate to the
+loader which one to use. We can use the `sample_label` argument to indicate which method
+to use. Pass one of "sample", for random sampling, or "all" to use all the available
+labels:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  type: "paired" # one of "paired", "unpaired" or "grouped"
+  labeled: true
+  sample_label: "sample" # one of "sample", "all" or None
+```
+
+In the case the `labeled` argument is false, the sample_label is unused, but still must
+be passed. Additionally, if the tensors in the files only have one label, irregardles of
+the `sample_label` argument, the data loader will only pass the one label to the
+network.
+
+##### Paired
+
+- `moving_image_shape`: (list, tuple) len 3, corresponding to (dim1, dim2, dim3) of the
```

Comment:
use a list or a tuple of three ints? more understandable ^^

Reply:
btw, these are from the docstring? maybe better to link to the function? as if we change the docstring, we have to change here as well (which is hard to detect/remember)

Reply:
If we change the functions from using list/tuples, we will need to modify the configs again, as the new configs will not be backcompatible in that case. So, maybe we should maintain the configs and for new changes away from the established in this config for 1.0.0 may need intermediate functions to adapt to this config? I can still link but these are my thoughts

Reply:
oh, i mean not for this single one only.

As here it seems that you copied the description of the args from multiple functions. I guess a relative link to the code would be easier to maintain?